src/ringct.rs

```diff
@@ -104,7 +133,37 @@ impl RingCtMaterial {
             })
             .collect::<Result<Vec<_>>>()?;
 
+        // Generate message to sign.
+        let mut msg: Vec<u8> = Default::default();
+        for t in true_input_public_keys.iter() {
+            msg.extend(t.to_bytes().as_ref());
+        }
```

Comment:
Careful, this is breaking our anonymous transactions. Any attacker can look at the first few bytes of a message to find the true public keys involved in the transaction.

Reply:
We want the order of public keys to be identical to the order of public keys in the final `RingCtTransaction`.